config/initializers/doorkeeper_openid_connect.rb

```diff
@@ -27,12 +27,36 @@
   end
 
   claims do
-    claim :preferred_username, :scope => :openid do |resource_owner, _scopes, _access_token|
+    claim :preferred_username, :response => [:id_token, :user_info] do |resource_owner, _scopes, _access_token|
```

Comment:
What was wrong with binding this to the `openid` scope?

Reply:
I guess it is not strictly required either way. https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims associates `preferred_username` with `profile` scope. However my tests with some providers show that it is returned when only `openid` scope is requested.

Contrary to the spec, they show up in id_token in addition to showing up in response to userinfo endpoint.

I was under the impression that you geared towards having `profile` scope with at least `profile` claim.

Reply:
Initially I hadn't realised that you had linked it to the `openid` scope so I was thinking we needed to add `profile` for anybody to see it but then I realised my error. You might be right that we should use the default though.

Reply:
I am a bit puzzled here too - some libraries ask for both "openid profile" scopes by default, and then the client app would add additional scopes to that. If we only have "openid" scope that would fail - the client app would have to do extra effort to remove "profile". I do not think this is mandated by the spec, but I guess most of the providers (if not all) offer both `openid` and `profile` scopes. They often put quite a bit of information from profile which is not required into id_token even though it is not required to be there (`name`, `email`, `preferred_name` often go to id_token...